packages/now-static-build/src/index.ts

```diff
@@ -493,8 +493,34 @@ export async function build({
       const extraOutputs = await readBuildOutputDirectory({ workPath });
 
       // No need to verify the dist dir if there are other output files.
-      if (!Object.keys(extraOutputs.output).length) {
-        validateDistDir(distPath);
+      if (extraOutputs.staticFiles) {
+        output = Object.assign(
+          {},
+          extraOutputs.staticFiles,
+          extraOutputs.functions
+        );
+      } else {
```

Comment:
Why is it if/else? Can the output only contain one or the other (static vs functions)?

Reply:
@styfle no, if `extraOutputs.staticFiles` is present, we'll use the files from there as output instead of checking `public` etc. for files.

For functions it's different, `public` could be used together with the functions from the `.vercel_build_output` directory, since those won't overwrite static assets.